src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,198 @@
+package duke.util;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.DukeException;
+import duke.command.ByeCommand;
+import duke.command.CheckCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.SaveCommand;
+import duke.command.TodoCommand;
+import duke.task.TaskList;
+
+/**
+ * The Parser class deals with making sense of the user command.
+ * @author Junyi
+ */
+public class Parser {
+
+    /* TaskList instance of tasks */
+    private final TaskList taskList;
+
+    /* Storage instance of Duke */
+    private final Storage storage;
+
+    /* Ui instance of Duke */
+    private final Ui ui;
+
+    private enum DukeCommand {
+        bye,
+        todo,
+        deadline,
+        event,
+        list,
+        mark,
+        unmark,
+        delete,
+        save,
+        check,
+        find
+    }
+
+    /**
+     * Constructor for Parser.
+     * @param taskList TaskList of Duke's tasks.
+     * @param storage Storage instance of Duke.
+     * @param ui Ui instance of Duke.
+     */
+    public Parser(TaskList taskList, Storage storage, Ui ui) {
+        this.taskList = taskList;
+        this.storage = storage;
+        this.ui = ui;
+    }
+
+    /**
+     * Validate the presence of args after a command.
+     * Does not precisely check validity of args for the command.
+     * @param cmd User command.
+     * @throws DukeException Exception thrown if no args were given.
+     */
+    private void validateNotEmptyArgs(String cmd) throws DukeException {
+        if (cmd.split(" ").length <= 1) {
+            throw new DukeException("You did not specify any details...");
+        }
+    }
+
+    /**
+     * Validate that index is within current set of tasks.
+     * @param index Index of the task, 0-indexed.
+     * @throws DukeException Exception thrown if index is invalid.
+     */
+    private void validateTaskIndex(Integer index) throws DukeException {
+        if (index < 0 || index >= taskList.size()) {
+            throw new DukeException("No such task!");
+        }
+    }
+
+    /**
+     * Returns a Command instance corresponding to the user input.
+     * @param cmd User input command.
+     * @return A command instance.
+     * @throws DukeException Thrown when user command has a format issue.
+     */
+    public Command parseUserCommand(String cmd) throws DukeException {
+        String firstCmd = cmd.split(" ")[0];
+        int taskIndex;
+        Command command;
+
+        try {
```

Comment:
Nice! This is a much better way than how I implemented my parser which was basically a long chain of if else statements 😄